src/hotspot/share/runtime/continuationFreezeThaw.cpp

```diff
@@ -473,9 +500,15 @@ void FreezeBase::unwind_frames() {
 
 template <typename ConfigT>
 template <bool chunk_available>
-freeze_result Freeze<ConfigT>::try_freeze_fast(intptr_t* sp) {
-  if (freeze_fast<chunk_available>(sp)) {
+freeze_result Freeze<ConfigT>::try_freeze_fast() {
+  if (chunk_available) {
+    freeze_fast_existing_chunk();
     return freeze_ok;
+  } else {
+    stackChunkOop chunk = allocate_chunk(_cont_size + frame::metadata_words);
+    if (freeze_fast_new_chunk(chunk)) {
```

Comment:
It seems odd that freeze_fast_existing_chunk() doesn't pass a chunk but freeze_fast_new_chunk does.  Should the allocate_chunk be inside of freeze_fast_new_chunk?

Reply:
The chunk is picked up from tail. Yes I agree it's not the way I liked.
But want to ship smaller pieces, since merging is really rough.